developer/js/tests/test-default-search-term-to-key.ts

```diff
@@ -0,0 +1,20 @@
+
+import 'mocha';
+import {assert} from 'chai';
+import { defaultWordform2Key } from '../dist/lexical-model-compiler/build-trie';
+
+
+describe('The default searchTermToKey() function', function () {
+  it('should lowercase and THEN normalize', function() {
+    // "İstanbul" has a DOTTED-I. This should be lowercased,
+    // then have its dot removed! 
+    assert.equal(defaultWordform2Key('İstanbul'), 'istanbul');
+    assert.equal(defaultWordform2Key('Diyarbakır'), 'diyarbakır');
```

Comment:
This unit test has opened up a can of philosophical worms for me. 🥫

(Side note: please include the Unicode values for the letters in the comment; helps avoid confusion.)

```suggestion
    // "İstanbul" has a U+0130 LATIN CAPITAL LETTER I WITH DOT ABOVE. 
    // This should be lowercased.
    assert.equal(defaultWordFormToKey('İstanbul'), 'istanbul');
    assert.equal(defaultWordFormToKey('Diyarbakır'), 'diyarbakır');
```

This comment is kinda orthographically misleading -- because in Turkish, `İ` lower cases to `i`; the dotless lower-case `ı` upper cases to `I` but we are not removing the dot in the assertion below it 😀. 

So the default normalization here is *not* normalizing U+0131 `ı` to U+0069 `i`, but it is normalizing U+0130 `İ` to U+0069 `i`? That means `'I' == 'İ' == 'i' != 'ı'` which is weird but is what Unicode specifies. But I don't think this is what we want.

There are two possibilities:
1. We want `'I' == 'İ' == 'i' == 'ı'` which makes sense for most Latin language orthographies except Turkic languages. However, we'll never see the U+0130/U+0131 versions except in Turkic languages(*) so this is kinda useless.
2. We are doing a Turkic language model, which means we want `'I' == 'ı', 'İ' == 'i'` but `'I' != 'i'`.

Given this is a _default_ function, we *could* ignore this but the unit test is going to be misleading and unhelpful to implementers either way as it stands!

(*) As far as I know; they may be used in other orthographies but Turkic (e.g. Turkish, Azerbaijani) is the well-known example.

Reply:
So I was half-thinking when I wrote this test. No, I don't think the default function should understand Turkic orthography; it's kind of an edge case in terms of Latin orthographies. The default should be a jack-of-all-orthographies, master-of-none kind of deal. I've updated the tests and comments to address my thoughts.